src/eduid_userdb/element.py

```diff
@@ -531,17 +546,28 @@ def _get_primary(self, elements):
         """
         if not elements:
             return None
-        res = [x for x in elements if x.is_primary is True]
+        verified = [x for x in elements if x.is_verified is True]
+        if len(verified) == 0:
+            try:
+                assert len([e for e in elements if e.is_primary]) == 0
+            except AssertionError:
+                raise PrimaryElementViolation('There are unconfirmed primary elements')
+            return None
+        res = [x for x in verified if x.is_primary is True]
```

Comment:
why not use self.verified here? maybe verified() should return a plain list instead of a new PrimaryElementList?